### PR TITLE
Redirect Warn to Output in the UI

### LINKED ISSUE #26736: Warning shows after UI input request on apply
### Terraform Version

```
v0.14.0-dev
```

### Terraform Configuration Files

```terraform
variable "cat" {
  default   = "cat"
  sensitive = true
}

resource "random_pet" "bar" {
  prefix = var.cat
}

terraform {
  experiments = [provider_sensitive_attrs]
}

```

### Debug Output

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # random_pet.bar will be created
  + resource "random_pet" "bar" {
      + id        = (known after apply)
      + length    = 2
      + prefix    = (sensitive)
      + separator = "-"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: 
Warning: Experimental feature "provider_sensitive_attrs" is active

  on main.tf line 52, in terraform:
  52:   experiments = [provider_sensitive_attrs]
```

### Expected Behavior

The warning should not displace the UI input, that is, your "yes" should be typed on the same line as the prompt.

### Steps to Reproduce

1. terraform init
2. terraform apply

### PR DESCRIPTION
**Pull request by jbardin**
The default cli Warn calls always write to the error writer (stderr by
default), however the output is intended to be viewed in the UI by the
user, rather than in a separate stream. Terraform also generally does
not consider warnings to be errors from the cli point of view, and does
not need to output the warning text to stderr.

By redirecting Warn calls to Output calls at the lowest level in the
main package, we can eliminate the chance that Warn and Output
messages are interleaved, while still allowing the internal `cli.Ui`
implementations to format `Warn` and `Output` calls separately.

Fixes #26736

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -36,6 +36,17 @@ const (
 	envTmpLogPath = "TF_TEMP_LOG_PATH"
 )
 
+// ui wraps the primary output cli.Ui, and redirects Warn calls to Output
+// calls. This ensures that warnings are sent to sudout, and are properly
```
Comment (alisdair) [inline main.go:40]:
```suggestion
// calls. This ensures that warnings are sent to stdout, and are properly
```

**Review by alisdair — APPROVED** [review]
🎉

**Comment by pselle** [comment]
🎉 Backport for 0.14?

**Comment by jbardin** [comment]
@pselle, Can you replicate the issue with the 0.14 branch? It does not happen for me, and the changes that made this consistently fail are only in 0.15.

**Comment by pselle** [comment]
Huh, then this PR doesn't close the linked issue, but the linked issue was fixed in some way at some point, according to this screenshot running on the current 0.14:

Screen Shot 2020-11-30 at 3 00 27 PM

My comment was because you'd attached it to the issue I'd filed, but I'll close that issue as fixed with the attached screenshot.

**Comment by jbardin** [comment]
@pselle 

I have a feeling that was generated from the master branch. The version string was not updated when 0.14 was cut, and you filed the issue before the string was set to `0.15.0`.

**Comment by pselle** [comment]
@jbardin Ahhh probably! Good memory 🐘

**Pull request merged**